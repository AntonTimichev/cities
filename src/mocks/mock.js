const o = 3;

export const leafletMock = {
  icon: jest.fn(),
  map() {
    return {
      setView: jest.fn(),
      remove: jest.fn(),
    };
  },
  marker() {
    return {
      addTo: jest.fn(),
    };
  },
  tileLayer() {
    return {
      addTo: jest.fn(),
    };
  },
  layerGroup() {
    return {
      addLayer: jest.fn(),
      addTo: jest.fn()
    };
  },
  featureGroup() {
    return {
      addTo: jest.fn()
    };
  }
};

export const OffersMock = [
  {
    bedrooms: 1,
    city: {
      location: {
        latitude: 38.77278,
        longitude: 4.44377,
        zoom: 16
      },
      name: `Amsterdam`
    },
    description: `This is good place`,
    goods: [`eat`, `meat`],
    host: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: 23,
      isPro: false,
      name: `Angelina`,
    },
    id: 1,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`
    ],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 42.78687,
      longitude: 4.373773,
      zoom: 16
    },
    maxAdults: 6,
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    price: 245,
    rating: 10,
    title: `Amazing and Extremely Central Flat`,
    type: `room`
  },
  {
    bedrooms: 4,
    city: {
      location: {
        latitude: 44.43737,
        longitude: 5.00434,
        zoom: 14
      },
      name: `Hamburg`
    },
    description: `I am happy to welcome you to my apartment in the city center!`,
    goods: [
      `Laptop friendly workspace`,
      `Towels`,
      `Breakfast`,
      `Washer`
    ],
    host: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: 27,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`
    ],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 42.47577,
      longitude: 4.1227,
      zoom: 15
    },
    maxAdults: 3,
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg`,
    price: 115,
    rating: 11,
    title: `The Joshua Tree House`,
    type: `apartment`
  },
  {
    bedrooms: 2,
    city: {
      location: {
        latitude: 44.58888,
        longitude: 4.00399,
        zoom: 12
      },
      name: `Paris`
    },
    description: `Three words: location, cosy and chic!`,
    goods: [
      `Washer`,
      `Breakfast`,
      `Air conditioning`
    ],
    host: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: 13,
      isPro: false,
      name: `Angelina`,
    },
    id: 3,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/10.jpg`
    ],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 59.3232,
      longitude: 4.99988,
      zoom: 18
    },
    maxAdults: 4,
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`,
    price: 199,
    rating: 8,
    title: `Waterfront with extraordinary view`,
    type: `house`
  },
];
