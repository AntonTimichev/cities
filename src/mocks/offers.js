import {getArrayCards, getRandomElements, getRandomInteger} from "../utils";

const getCardData = () => {
  return {
    id: Math.random().toString(16).slice(2, 8),
    name: getRandomElements(names, 1)[0],
    src: `img/room.jpg`,
    isPremium: !!getRandomInteger(0, 1),
    price: getRandomInteger(60, 120),
    inBookMark: !!getRandomInteger(0, 1),
    roomType: getRandomElements(roomTypes, 1)[0]
  };
};

const roomTypes = [
  `Apartment`,
  `Private room`
];

const names = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`,
];

export const offers = getArrayCards(getCardData, 5);
