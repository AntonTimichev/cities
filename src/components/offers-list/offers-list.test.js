import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from 'react-router-dom';

import OffersList from "./offers-list.jsx";
import {OffersMock} from "../../mocks/mock";

it(`OffersList is rendered correctly`, () => {
  const tree = renderer.create(<MemoryRouter>
    <OffersList
      offers={OffersMock}
      activeItem={0}
      onItemClick={jest.fn()}
      onOfferImgClick={jest.fn()}
    />
  </MemoryRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
