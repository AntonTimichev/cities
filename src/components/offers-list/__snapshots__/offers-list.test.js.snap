// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList is rendered correctly 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    style={
      Object {
        "opacity": "",
      }
    }
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://es31-server.appspot.com/six-cities/static/hotel/15.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            245
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In
             bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "93%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Amazing and Extremely Central Flat
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    style={
      Object {
        "opacity": "",
      }
    }
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://es31-server.appspot.com/six-cities/static/hotel/9.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            115
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To
             bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "93%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          The Joshua Tree House
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    style={
      Object {
        "opacity": "",
      }
    }
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://es31-server.appspot.com/six-cities/static/hotel/3.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            199
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In
             bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "93%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Waterfront with extraordinary view
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;
